compiler/runtime/Runtime.hpp

```diff
@@ -170,31 +170,66 @@ enum TR_RuntimeHelper
 class TR_RuntimeHelperTable
    {
 public:
-   void* getAddress(TR_RuntimeHelper h)
-      {
-      return h < TR_numRuntimeHelpers ? _helpers[h] : (void *) (uintptr_t) 0xDEADB00F;
-      }
+
+   /**
+    * \brief
+    *
+    * For helper function, this returns the raw entry address. If the runtime
+    * helper's corresponding linkage is not a function, return the helper as data.
+    */
+   void* getAddress(TR_RuntimeHelper h);
+
+   /**
+    * \brief
+    *
+    * Looks up the given helper index and returns the environment pointer, if there is any.
+    * The environment pointer of a function is applicable to PPC AIX and z/OS.
+    */
+   void* getEnvironment(TR_RuntimeHelper h);
+
+   /**
+    * \brief
+    *
+    * Returns the linkage type of the given helper index.
+    */
    TR_LinkageConventions getLinkage(TR_RuntimeHelper h)
       {
       return h < TR_numRuntimeHelpers ? _linkage[h] : TR_None;
       }
-   // Linkage convention is essential when calling a helper
-   // For example, on X86, there are private linkage, System V ABI, fastcall, cdecl, etc.
+
+   /**
+    * \brief
+    *
+    * Sets the address and linkage convention.
+    *
+    * Linkage convention is essential when calling a helper
+    * For example, on X86, there are private linkage, System V ABI, fastcall, cdecl, etc.
+    */
    void setAddress(TR_RuntimeHelper h, void * a, TR_LinkageConventions lc = TR_Helper)
       {
-      _helpers[h] = translateAddress(a);
+      _helpers[h] = a;
       _linkage[h] = lc;
       }
 
+   /**
+    * \brief
+    *
+    * Stores the given pointer as a helper constant.
+    */
    void setConstant(TR_RuntimeHelper h, void * a)
       {
       _helpers[h] = a;
+      _linkage[h] = TR_None;
       }
 
 private:
-   // translate address is to allow each platform converting a C function pointer
-   // to an address callable by assembly, which is essential for P and Z.
+   /**
+    * \brief
+    * Translate address is to allow each platform converting a C function pointer
+    * to an address callable by assembly, which is essential for P and Z.
+    */
```

Comment:
Same as above, "P and Z" might go stale.

Reply:
Fixed.